utest/utest.h

```diff
@@ -12,7 +12,8 @@
 #include "boost/filesystem/path.hpp"
 #include "boost/filesystem/operations.hpp"
 
-typedef PointMatcher<float> PM;
+typedef float NumericType;
```

Comment:
Why that extra `typedef`?

Reply:
To be able to use the numeric type of the DataPoints in the tests too. For example:

```
static inline Eigen::Transform<NumericType, 2, Eigen::Affine> buildUpTransformation2D(const Eigen::Matrix<NumericType, 2, 1>& translation,
                                                                                      const Eigen::Rotation2D<NumericType>& rotation,
                                                                                      const NumericType scale = 1.0)
{
    const Eigen::Transform<NumericType, 2, Eigen::Affine> transformation =
        Eigen::Translation<NumericType, 2>(translation) * rotation * Eigen::Scaling(scale);
    return transformation;
}
```

In the above function all types are dependent on the NumericType of the DataPoints object.


This is a small step, but long term we could further look into extending this, to run the unit tests with different data types, using e.g. typed tests from gtest.